mathics/builtin/manipulate.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+from mathics.core.expression import String
+from mathics import settings
+from mathics.core.evaluation import Evaluation
+
+from mathics.builtin.base import Builtin
+from mathics.core.expression import Expression, Symbol, Integer, from_python
+
+import six
+
+try:
+    from ipykernel.kernelbase import Kernel
+    _jupyter = True
+except ImportError:
+    _jupyter = False
+
+try:
+    from ipywidgets import (IntSlider, FloatSlider, ToggleButtons, Box, DOMWidget)
+    _ipywidgets = True
+except ImportError:
+    # fallback to non-Manipulate-enabled build if we don't have ipywidgets installed.
+    _ipywidgets = False
+
+
+"""
+A basic implementation of Manipulate[]. There is currently no support for Dynamic[] elements.
+This implementation is basically a port from ipywidget.widgets.interaction for Mathics.
+"""
+
+def _interactive(interact_f, kwargs_widgets):
+    # this is a modified version of interactive() in ipywidget.widgets.interaction
+
+    container = Box(_dom_classes=['widget-interact'])
+    container.children = [w for w in kwargs_widgets if isinstance(w, DOMWidget)]
+
+    def call_f(name=None, old=None, new=None):
+        kwargs = dict((widget._kwarg, widget.value) for widget in kwargs_widgets)
+        try:
+            interact_f(**kwargs)
+        except Exception as e:
+            container.log.warn("Exception in interact callback: %s", e, exc_info=True)
+
+    for widget in kwargs_widgets:
+        widget.on_trait_change(call_f, 'value')
+
+    container.on_displayed(lambda _: call_f(None, None, None))
+
+    return container
+
+
+def _strip_namespace(name):
```

Comment:
It looks like this function does the same thing as `mathics.core.expression.strip_context` - maybe we can use that here instead?

I think it makes sense to use this in `PatternDispatcher.dispatch` (because we need valid Python argument names), but for generating the labels `Definitions.shorten_name` might be a better choice - this only shortens the name if it's accessible via `$ContextPath`. I don't have Mathematica around to check but I'd guess this is what it does.